src/CircuitBreaker.ts

```diff
@@ -62,12 +68,15 @@ export class CircuitBreaker extends EventEmitter {
     }
 
     async _evaluate() {
-        this._evaluatingPromiseHook = Promise.allSettled(this._probes.map(p => p.check()));
+        this._cbError = false;
+        this._evaluatingPromiseHook = Promise.allSettled(this._probes.map(p => p.check().catch(() => {
+            this._cbError = true;
+            return 'failed';
```

Comment:
`check()` returns a void, so the `catch` callback should also return `void` (i.e. nothing)...

Reply:
the catch block can return any value, not necessarily `void`. The returned value 'failed' is a way to handle the error and provide a default value when a probe check fails. It is indeed not used in the current code, but I thought it could be useful if we ever decide to handle failed probes differently in the future.

Reply:
it is not about being "allowed" to return a value, but about type consistency : `check()` is "void" Promise, so catch should also return the same type.

if in the future we change check to return some sort of result, we will change the catch block to return a matching result. But no point introducing some magic string at the moment.

Reply:
https://github.com/scality/breakbeat/pull/11/commits/d9bd1f205d1494f6f1fafae4a44b5c5b8994373c#diff-359e4042f12fa2b33c080ef07d2ff300cdd9137a3f55c0dc89b5d823527da29fR71-R73